### PR TITLE
A completely New look

### PR DESCRIPTION
**Pull request by thesuss**
PT Story:  https://www.pivotaltracker.com/story/show/161343611

Changes proposed in this pull request:
1.  New design based on Bootstrap
2. Much less Css code

Ready for review and help <3

### CODE REVIEW

**Comment by weedySeaDragon** [comment]
FYI:  the test are failing on associations where `belongs_to` is `optional`

The Gemfile.lock uses the up-to-date `shoulda-matcher` gem which does a more restrictive and accurate test (match) per Rails 5 and `belongs_to ...., optional:true`. 

I'm looking at this now.

**Comment by thesuss** [comment]
YEY, Only codeclimate is angry now.

**Comment by patmbolger** [comment]
@thesuss Overalls it looks good - clean and crisper.

My comments:

1. Home page - see screenshot:

1  Home


2. Home page:

2  Home

(note that this is problem with all of the "Hide" buttons)

3. Admin view of apps:

3  Admin view of apps

4. User View:

4a  User View

In general, this design does not make effective use of all the white space to the right of the initial data items.  I'd prefer something more like the current view (from the POV of effective use of real estate):

4b  Current User View

5. Company View:

5 Company View

6. Waiting for reason admin view:

6  Waiting for reason

7. Categories - admin view (Swedish):

7  Categories

8. Categories - admin view (English) (possible translation problem?):

8  Categories (English)

9. All Users - items-per-page control is broken:

9  All Users

10. Edit profile page (English):

10  Edit profile

11. Edit member status:

11  Edit Member Status

This should be styled similar to the branding-status modal (see below):

  - Title should be left-justified
  - Radio buttons need more separation
  - Date fields done like branding edit
  - Notes field done like branding edit

12. Edit Branding status (looks great, included for comparison):

12  Edit Branding Status

**Review by patmbolger — COMMENTED** [review]
Please see comments.

**Comment by thesuss** [comment]
Links seem/are inconsistent because: 
- Bold in tables 
- Border bottom outside of tables. 
border bottom in tables make it look *very* cluttered and hard to read anything (imo). 
@weedySeaDragon  @rmzse  what do you think?

**Comment by thesuss** [comment]
4 User view - I agree, and there is more I want to do but can think that can be a separate because there is SO MUCH code on that page messing with any and all design thoughts ;) .
I dont think I've actually changed much there, so it must be bootstrap not supporting the classes there anymore,

**Comment by thesuss** [comment]
Ok... lets see how many cucumbers disagree...

**Comment by thesuss** [comment]
@patmbolger 
1 - agree but can be separate, is tricky
2 - done, but want to change them completely
3 - same
4 - agree but can be separate, is tricky
5 - I want to change the labels to fa-icons. Especially when social media comes along. 
Also see separate comment re: links. 
6 - fixed
7 - fixed
8 - Design is fixed
9 - fixed
10 - Not fixed 
11 - Fixed
12 - Agree, its pretty.

**Comment by thesuss** [comment]
the (i) is missing after the upcoming payment dates.
 Probably a glyph that needs to be a fa(icon) thing. I can't find them :(

**Comment by weedySeaDragon** [comment]
In this PR we should just be addressing:
1. what users, visitors and members see (not admin)
2. **errors** (not design improvements)

Any changes for admin views should be a separate PR so that we can get this one finished and committed. Likewise, design improvements need to be discussed and addressed in a separate PR.

(more later; all I have time for ATM)

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -164,4 +164,45 @@ def yes_no_span(boolean_value)
...
+  #
+  #
+  # @param title [String] - the
+  # @param fa_icon_group [String] - one of the main FontAwesome groups. default is 'fas'
+  # @param fa_icon [String] - the FontAwesome icon to use, _without_ the leading 'fa'. default is 'info-circle'
+  #   The leading 'fa-' is added automatically
+  #
+  # @return [String] - the html safe string for the entire <span>
+  #
+  def fas_tooltip(title, fa_icon_group: 'fas', fa_icon: 'info-circle')
```
Comment (thesuss) [inline app/helpers/application_helper.rb:206]: ❤️

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -164,4 +164,45 @@ def yes_no_span(boolean_value)
...
+  # Example: fas_tooltip(I18n.t('share_on_facebook'), fa_icon_group: 'fab', fa_icon: 'facebook')
+  #   will return
+  #    '<span class="i fab fa-facebook"
+  #        title=""
+  #        data-original-title="Share on Facebook">
+  #        data-toggle="tooltip"
+  #     </span>'
+  #
+  #
+  # @param title [String] - the
```
Comment (thesuss) [inline app/helpers/application_helper.rb]: here may be words missing?

Comment (weedySeaDragon) [reply app/helpers/application_helper.rb]: yup - good catch. :-)  I'll finish that up in a bit.

**Comment by weedySeaDragon** [comment]
<sigh> I need to fix the payments_helper.

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]
### UI errors that need to be fixed:
(screenshots are attached after the text list)

### A) 4 places where the bottom links need CSS classes so they appear as buttons
1. User edits profile
2. Member edits company
3. Member adds new address
4. Member edits address

### B) Member creates new company: modal needs title alignment and input box fix

---
### Screenshots

**A) 1. User edits profile -- ignore those green key icons. it's a plugin I have**
newLook-todo-user-edit-profile-button

**A) 2. Member edits company**
newLook-todo-edit-co-buttons

**A) 3. Member adds new address**

newLook-todo-new-addr-button

**A) 4. Member edits address**

newLook-todo-edit-address


**B) Member creates new company modal**
newLook-todo-create-co-alignment2

**Comment by thesuss** [comment]
I disagree that the bottom links **need** to be buttons. 
Delete something could be a button to make it clear it's an action rather than just getting around. 
 - - Aaaaand if Patrick, Rodrigo and Jenny all agree they should be buttons that's an easy fix for later. 

Member creates new company modal - Will fix.

**Comment by weedySeaDragon** [comment]
You have my feedback about the bottom actions - I think they should be buttons. 

New company modal looks good!

**Review by weedySeaDragon — APPROVED** [review]

**Comment by patmbolger** [comment]
Referencing my numbers ....

7. - still looks broken:

Screen Shot 2019-03-20 at 5 27 49 PM

8. Title still messed up:

Screen Shot 2019-03-20 at 5 29 32 PM

The others that you worked on look fixed.

I think that the visitor/user/member home page (screenshot 1) should be fixed.  It looks very jumbled and does not present us well to the users.

Also, I agree that this PR can exclude fixes for admin-only views.

But, I do think we need a short style guide documenting how we render various items, such as links, column headers, etc. then work to adopt those guidelines (perhaps after this PR is deployed).

**Comment by thesuss** [comment]
@patmbolger 7 scroll down a bit and you'll see why :)  Only 2 have descriptions. 
Lines will be prettier when view/Edit/remove are icons not words

and 8: It's your language :) 
I will go though the language in Swedish with Jenny after this, and fix a bunch of stuff. If it bugs you I will try to fix also this one :)

**Comment by thesuss** [comment]
I do also think that the visitor/user/member home page (screenshot 1) should be fixed. 

I know how I want it to look and am very near knowing what to do - but not qute there. I think it can be slightly jumbled from thursday to probably monday. I think by then I'll have a piece done we can all agree on.

I would like to get this huge thing merged and do fixes in smaller pieces.  
I'm sure we will find more bits and pieces needing fixing day by day after, some more important than others. That happens with a complete overhaul. 

We don't actually need to deploy tomorrow but can let it hang in heroku and check it out there also. And then I can still do the landing in a separate PR

**Pull request merged**